packages/modules/viewer-react/src/components/error/ErrorBoundary.tsx

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import type { ErrorPageProps, ErrorPageType } from "@itwin/itwinui-react";
+import { ErrorPage } from "@itwin/itwinui-react";
+import React, { Component, Fragment } from "react";
+
+interface Props {
+  eventTitle?: string;
+}
+
+export class ErrorBoundary extends Component<
+  Props,
+  {
+    fallback: boolean;
+    error: Error;
+  }
+> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      fallback: false,
+      error: new Error(),
+    };
+  }
+
+  static getDerivedStateFromError(e: Error): {
+    fallback: boolean;
+    error: Error;
+  } {
+    return {
+      fallback: true,
+      error: e,
+    };
+  }
+
+  override render(): JSX.Element {
+    if (this.state.fallback) {
+      let errorType: ErrorPageType = "generic";
+      let errorMessage: ErrorPageProps["errorMessage"] = (
+        <>
+          We can't find the iModel that you are looking for or it does not
+          exist. Visit the iModel HUB or contact our support team.
+        </>
+      );
+
+      if (this.state.error.message) {
```

Comment:
what is the type of error.message?
i question the error messages here, i guess these are copied from the current fallback we use, but these were pulled out of design review. i wonder if we should just use the message from the error itself.

@kckst8  thoughts?

Reply:
@kckst8 , would you be ok if we always used the generic error page: https://itwin.github.io/iTwinUI-react/?path=/story/core-errorpage--generic and the message would just be different and populated by the actual error message?

The current errors thrown don't have http status code unfrotunately, and Id rather not we go through all the possible erros here and map them to the right status code. https://github.com/iTwin/imodels-clients/blob/main/clients/imodels-client-management/src/base/interfaces/IModelsErrorInterfaces.ts

I'll need to talk to Austeja because i dont like they don't report the status code, and they just report their own enum. Feel like all our clients should've standardized error handling

Reply:
So this is what an error page would look like with a generic title and svg, but with the error message populated by the actual error message: 
![image](https://user-images.githubusercontent.com/51834103/171234332-643edc9a-0a56-4828-9893-1a9073e31ea4.png)
Ideally, we'd want to have a custom title and svg that maps to each unique error code though.